content/assignments/techjobs-console/_index.md

```diff
@@ -0,0 +1,477 @@
+
+In the `JobData` class, find the method `FindByValue()`. This method has been outlined
+for you but contains none of the code needed to work (you should leave the `LoadData()` call as the first line of the method, however). Here are a few observations:
+
+1. The code that you write should not contain duplicate jobs. So, for example, if a listing has position type "Web - Front End" and name "Front end web dev" then searching for "web" should not include the listing twice.
+1. As with `PrintJobs()`, you should write your code in a way that if a new column is added to the data, your code will automatically search the new column as well.
+1. You should NOT write code that calls `FindByColumnAndValue()` once for each column. Rather, utilize loops and collection methods as you did above.
+1. You *should*, on the other hand, read and understand `FindByColumnAndValue()`, since your code will look similar in some ways.
+
+You'll need to call `FindByValue()` from somewhere in `RunProgram()`. We'll
+leave it up to you to find where. You might have noticed that when you
+try to search all columns using the app, a message is printed, so that
+is a good clue to help you find where to place this new method call.
+Once you find where to call your new method, you can *Run* the program
+again to test your code.
+
+### Make Search Methods Case-Insensitive
+
+You've completed your first two tasks!
+
+Let's assume you demonstrated the updated application for the Company Team, and
+they noticed a feature that could be improved. When searching for jobs with
+the skill `"JavaScript"` some results were missing (e.g. the Watchtower
+Security job on line 31 of the CSV file). The search methods turn out to be
+case-sensitive, so they treat `"JavaScript"` and `"Javascript"` as different
+strings.
+
+The Company Team strongly requested that this needs to be fixed, and of course
+you told them that you are up to the task.
+
+Here are some questions to ask yourself as you get started:
+
+1. Which methods are called when searching?
+1. How is the user's search string compared against the values of fields of the job `Dictionary` objects?
+1. How can you make this comparison in a way that effectively ignores the case of the strings?
+1. How can you do this *without* altering the capitalization of the items in `AllJobs` so that the data gets printed out the same way that it appears in `job_data.csv`?
+
+<!-- TODO: Add reference to string methods section in string chapter -->
+
+You might find it useful to review the String methods listed in the
+chapter.
+
+When this task is completed, you're done!
+
+## Sanity Check
+
+Before submitting, make sure that your application:
+
+1. Prints each field of a job when using search functionality, and when listing all columns. If there are no search results, a descriptive message is displayed.
+1. Allows the user to search for a string across all columns.
+1. Returns case-insensitive results.
+
```

Comment:
Would we want to say something about passing the tests here?

Reply:
Added it.

Reply:
Thanks.